bindings/java/src/org/sleuthkit/datamodel/HostManager.java

```diff
@@ -0,0 +1,170 @@
+			throw new IllegalArgumentException("Host name is required.");
+		}
+
+		CaseDbConnection connection = transaction.getConnection();
+
+		// First search for host by name
+		Optional<Host> host = getHost(name, connection);
+		if (host.isPresent()) {
+			return host.get();
+		}
+
+		// could'nt find it, create a new host
+		return createHost(name, connection);
+	}
+	
+	/**
+	 * Get host with given name. 
+	 * 
+	 * @param name Host name to look for. 
+	 * @param connection Database connection to use.
+	 * 
+	 * @return Optional with host.  Optional.empty if no matching host is found.
+	 * 
+	 * @throws TskCoreException 
+	 */
+	Optional<Host> getHost(String name, CaseDbConnection connection ) throws TskCoreException {
+		
+		String queryString = "SELECT * FROM tsk_hosts"
+							+ " WHERE LOWER(name) = LOWER('" + name + "')";
+
+		try (Statement s = connection.createStatement();
+				ResultSet rs = connection.executeQuery(s, queryString)) {
+
+			if (!rs.next()) {
+				return Optional.empty();	// no match found
+			} else {
+				return Optional.of(new Host(rs.getLong("id"), rs.getString("name")));
+			}
+		} catch (SQLException ex) {
+			throw new TskCoreException(String.format("Error getting host with name = %s", name), ex);
+		}
+	}
+	
+
+	/**
+	 * Create a host with given name.
+	 * 
+	 * @param name Host name.
+	 * @param connection Database connection to use. 
+	 * @return Newly created host.
+	 * @throws TskCoreException 
+	 */
+	private Host createHost(String name, CaseDbConnection connection) throws TskCoreException {
+		db.acquireSingleUserCaseWriteLock();
+		try {
+			String hostInsertSQL = "INSERT INTO tsk_hosts(name) VALUES (?)"; // NON-NLS
+			PreparedStatement preparedStatement = connection.getPreparedStatement(hostInsertSQL, Statement.RETURN_GENERATED_KEYS);
+
+			preparedStatement.clearParameters();
+			preparedStatement.setString(1, name);
+
+			connection.executeUpdate(preparedStatement);
+
+			// Read back the row id
+			try (ResultSet resultSet = preparedStatement.getGeneratedKeys();) {
+				if (resultSet.next()) {
+					return new Host(resultSet.getLong(1), name); //last_insert_rowid()
+				} else {
+					throw new SQLException("Error executing  " + hostInsertSQL);
+				}
+			}
+		} catch (SQLException ex) {
+			LOGGER.log(Level.SEVERE, null, ex);
+			throw new TskCoreException(String.format("Error adding host with name = %s", name), ex);
+		} finally {
+			db.releaseSingleUserCaseWriteLock();
+		}
+	}
+	
+	/**
+	 * Get all hosts.
+	 * 
+	 * @return Collection of hosts.
+	 * @throws TskCoreException 
+	 */
+	Set<Host> getHosts() throws TskCoreException {
+		String queryString = "SELECT * FROM tsk_hosts";
+
+		Set<Host> hosts = new HashSet<>();
+		try (CaseDbConnection connection = this.db.getConnection();
+				Statement s = connection.createStatement();
+				ResultSet rs = connection.executeQuery(s, queryString)) {
+
+			while (rs.next()) {
+				hosts.add(new Host(rs.getLong("id"), rs.getString("name")));
```

Comment:
I think we need to pass the status column value to the Host constructor as well.